grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/rule/InferenceRule.java

```diff
@@ -291,6 +291,8 @@ public Unifier getUnifier(Atom parentAtom) {
         }
         //case of match all relation atom
         else{
+            // TODO: is this cast always safe?
```

Comment:
@kasper-piskorski can you answer this?

Reply:
I think it's not safe :( Instead I should make the addType method Atom specific.